### DISCUSSION TITLE
Workspace not using target specified

### DISCUSSION THREAD

**Discussion by samcarey** [bevyengine/bevy#3263] [Q&A] [answered]
I'm having trouble building bevy main with bevy_webgl2 in Cargo workspace. Somehow, the dependencies change to include `ash` whenever I switch to a workspace. I think `ash` is incompatible with web/wasm32.

I have a binary project with the following Cargo.toml:
```toml
[package]
name = "test"
version = "0.1.0"
edition = "2021"

[dependencies]
bevy = { git = "https://github.com/bevyengine/bevy.git", default-features = false }
bevy_webgl2 = { git = "https://github.com/billyb2/bevy_webgl2.git" }
```
and an empty main function. The layout is:
```
src
  main.rs
Cargo.toml
```

This will compile when I run `cargo build --target wasm32-unknown-unknown`.

However, if I move the `src` directory and `Cargo.toml` to a directory and add a root `Cargo.toml` (creating a workspace):
```toml
[workspace]
members = ["test"]
```
with layout:
```
test
  src
    main.rs
  Cargo.toml
Cargo.toml
```
and then run `cargo build --target wasm32-unknown-unknown`, I get this error:
```
error[E0432]: unresolved import `libloading::Library`
 --> C:\Users\sccarey\.cargo\registry\src\github.com-1ecc6299db9ec823\ash-0.33.3+1.2.191\src\entry_libloading.rs:3:5
  |
3 | use libloading::Library;
  |     ^^^^^^^^^^^^^^^^^^^ no `Library` in the root

error[E0425]: cannot find value `LIB_PATH` in this scope
  --> C:\Users\sccarey\.cargo\registry\src\github.com-1ecc6299db9ec823\ash-0.33.3+1.2.191\src\entry_libloading.rs:81:28
   |
81 |         Self::with_library(LIB_PATH)
   |                            ^^^^^^^^ not found in this scope
```

Using cargo tree, I've determined that `ash` is being included as a dependency in the latter case, but not the former.
`ash` is from `wgpu-hal`, feature `vulkan`.
`vulkan` is being activated from `wgpu-core`, but the only places it's included are not supposed to build under the `wasm32` target, which I've activated with the `--target` arg in my build command. So, it seems to me that when using a workspace, the target specification is being ignored.

**Comment by NiklasEi** [comment, marked as answer]
One thing with workspaces is that [they still seem to have resolver v1 as their default](https://doc.rust-lang.org/stable/edition-guide/rust-2021/default-cargo-resolver.html#details). I've seen other issues using Bevy main get fixed by adding `resolver = "2"` to the workspace `Cargo.toml`. Does that help here?

**Reply by samcarey** [reply]
That did it! Thanks!
